Changes for different petfriends pricelist

pascalckunz opened the pull request:
add methods to filter pricelists with timestamps
delphin uploads 2 files with a timestamp each day.
these files have to be taken from the specific directory
the old files have to be deleted
the functions for s3 and url download are not neccessary

Commit:
change pricefield name and delimiter

Commit:
bump version

Commit:
add methods to filter pricelists with timestamps

delphin uploads 2 files with a timestamp each day.
these files have to be taken from the specific directory
the old files have to be deleted
the functions for s3 and url download are not neccessary

pascalckunz committed:
Merge branch 'master' into changes-for-different-petfriends-pricelist

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
         return $this->_import;
     }
+
+    public function getCurrentFile(): array
```
Why is it in singular? You return an array of file names here.

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+    public function prepareFileNames($originalFilenames)
+    {
+        if (empty($originalFilenames)) {
+            return;
```
```suggestion
            return null;
```
Do not return nothing.
This prevents an error like "try to use return value of void function".

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+    /**
+     * @param $originalFilenames
+     */
+    public function prepareFileNames($originalFilenames)
```
```suggestion
    public function prepareFileNames($originalFilenames): ?array
```

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+        if (empty($originalFilenames)) {
+            return;
+        }
+        if (!empty($originalFilenames)) {
+            foreach ($originalFilenames as $name) {
+                $filenames[$name] = substr($name, strripos($name, '/') + 1);
+            }
+        }
```
```suggestion
        foreach ($originalFilenames as $name) {
            $filenames[$name] = substr($name, strripos($name, '/') + 1);
        }
```
It is never empty after the first `if`.

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+    /*
+     * use $hasSuffix = 1 for list with suffix
+     */
+    public function cleanFileNames(array $fileNames, int $hasSuffix = 0): string
```
```suggestion
    public function cleanFileNames(array $fileNames, bool $hasSuffix = false): string
```
Why not bool?

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+    }
+
+    /*
+     * use $hasSuffix = 1 for list with suffix
```
```suggestion
     * use $hasSuffix = true for list with suffix
```

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+    public function cleanFtpFiles(array $fileNames): void
+    {
+        foreach ($fileNames as $filename) {
+            $this->deleteFtpFiles($this->getFtpFilesLocation() . $filename);
```
What if there is no trailing `/` in `'ambimax_priceimport/erp_import_options/file_sftp_path'`?

pascalckunz replied:
then it does not work

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+        }
+    }
+
+    public function deleteFtpFiles($filePath): void
```
```suggestion
    public function deleteFtpFile($filePath): void
```

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+
+    }
+
+    protected function getFtpFileNames(): string
```
Is there a way, where the return value has more than one file name? Use singular in the whole file for this name, if not.

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Helper/Data.php:
```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+        if (!str_starts_with($fileName, $fileNameSetting)) {
+            return false;
+        }
+        if (!str_starts_with($fileName, $fileNameSetting . $this->_listsuffix)) {
```
Is it really something like `price-list-file-name.csv-fwpf-`? Seems weird to me.

pascalckunz replied:
its artikelstamm-fwpf-20210622010002

pascalckunz replied:
you only use the filename start in the administration fields

Wysselbie replied:
Aaaahhh

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php:
```diff
@@ -64,42 +64,41 @@ public function setPriceData($data, $doMapping = true)
-            if (!$sku = $row['sku']) {
-                continue;
-            }
+            $data = array();
+            $columns = null;
+            $map = array_flip($this->_map);
```
Is it right, that `$data` will be reset every iteration? You return it with this method. Here only the data of the last iteration would be returned.
And what about multiple entries with the same SKU in different files?
And `$map` does not have to be in the `foreach`, too, in my opinion.

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php:
```diff
@@ -109,9 +108,9 @@ public function loadCsvData($fileLocation)
+    public function getLocalFilePath($fileName)
     {
-        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_path');
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_path') . $fileName;
```
What about a missing trailing `/` in `'ambimax_priceimport/erp_import_options/file_path'`?

Wysselbie commented on src/app/code/community/Ambimax/PriceImport/Model/ErpImport.php:
```diff
@@ -148,7 +147,7 @@ protected function _downloadSftpFile()
         $io->checkAndCreateFolder(dirname($destination));
 
-        $connectionString = str_replace(array_keys($options), $options, 'sftp://{username}:{password}@{host}{path}');
+        $connectionString = str_replace(array_keys($options), $options, 'sftp://{username}:{password}@{host}{path}' . $fileName);
```
What about a missing trailing `/` in `{path}`?

Commit:
multiple code quality changes

changed plurals and singulars for different function
added descriptions for Backend fields

Commit:
Merge branch 'changes-for-different-petfriends-pricelist' of https://github.com/ambimax/magento-module-ambimax-priceimport into changes-for-different-petfriends-pricelist

Commit:
refactored loadCsvData function
